pkg/apis/triggers/v1alpha1/trigger_binding_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+/*
+Copyright 2019 The Tekton Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check that TriggerBinding may be validated and defaulted.
+//var _ apis.Validatable = (*TriggerBinding)(nil)
+
+// +k8s:deepcopy-gen=true
+type TriggerBindingSpec struct {
+	Templates []TriggerTemplateRef `json:"templates,omitempty"`
```

Comment:
IIRC, we mentioned that we would support references as well as embedded TriggerTemplates?

Reply:
yep fo sho! adding embedding later on might be easier tho (and starting with just references)

Reply:
Yes exactly ^ I thought we could add references now and then add the embedded handling separately, to make this change smaller and to (maybe?) easier to reason about what is doing what. Im up for adding support for embedding templates now if thats betterer.